tests/testsuite/pkgid.rs

```diff
@@ -311,75 +307,27 @@ fn pkgid_json_message_metadata_consistency() {
     assert_e2e().eq(pkgid, str!["path+[ROOTURL]/foo#0.5.0"]);
 
     p.cargo("check --message-format=json")
-        .with_json(
-            &r#"
-{
-  "reason": "compiler-artifact",
-  "package_id": "$PKGID",
-  "manifest_path": "[..]",
-  "target": "{...}",
-  "profile": "{...}",
-  "features": [],
-  "filenames": "{...}",
-  "executable": null,
-  "fresh": false
-}
-
-{
-  "reason": "build-script-executed",
-  "package_id": "$PKGID",
-  "linked_libs": [],
-  "linked_paths": [],
-  "cfgs": [],
-  "env": [],
-  "out_dir": "[..]"
-}
-
-{
-  "manifest_path": "[..]",
-  "message": "{...}",
-  "package_id": "$PKGID",
-  "reason": "compiler-message",
-  "target": "{...}"
-}
-
-{
-  "reason": "compiler-message",
-  "package_id": "$PKGID",
-  "manifest_path": "[..]",
-  "target": "{...}",
-  "message": "{...}"
-}
-
-{
-  "reason": "compiler-artifact",
-  "package_id": "$PKGID",
-  "manifest_path": "[..]",
-  "target": "{...}",
-  "profile": "{...}",
-  "features": [],
-  "filenames": "{...}",
-  "executable": null,
-  "fresh": false
-}
-
-{
-  "reason": "build-finished",
-  "success": true
-}
-            "#
-            .replace("$PKGID", pkgid),
-        )
+            .with_stdout_data(str![[r#"
+{"executable":null,"features":[],"filenames":"{...}","fresh":false,"manifest_path":"[ROOT]/foo/Cargo.toml","package_id":"path+[ROOTURL]/foo#0.5.0","profile":"{...}","reason":"compiler-artifact","target":"{...}"}
+{"cfgs":[],"env":[],"linked_libs":[],"linked_paths":[],"out_dir":"[ROOT]/foo/target/debug/build/foo-[HASH]/out","package_id":"path+[ROOTURL]/foo#0.5.0","reason":"build-script-executed"}
+{"manifest_path":"[ROOT]/foo/Cargo.toml","message":"{...}","package_id":"path+[ROOTURL]/foo#0.5.0","reason":"compiler-message","target":"{...}"}
+{"manifest_path":"[ROOT]/foo/Cargo.toml","message":"{...}","package_id":"path+[ROOTURL]/foo#0.5.0","reason":"compiler-message","target":"{...}"}
+{"executable":null,"features":[],"filenames":"{...}","fresh":false,"manifest_path":"[ROOT]/foo/Cargo.toml","package_id":"path+[ROOTURL]/foo#0.5.0","profile":"{...}","reason":"compiler-artifact","target":"{...}"}
+{"reason":"build-finished","success":true}
+
+"#]].json_lines())
```

Comment:
Please let me know if there's a better approach to working with prettified json.

Reply:
Probably better to not do `json_lines` at this time.  We want to find a better solution for pretty printing it.

Reply:
I've reverted this change and allowed the deprecated `with_json` for now.